### PR TITLE
setting up Eslint and Prettier for code formating 

### LINKED ISSUE #32: Add code format / linting on code commit event in the repo
## Description
So far we do not have a code formatting / linting tool integrated into the project. We need to integrate some tool like eslint and also configure it to run on git commit hook so that only those commits will be accepted which have acceptable code formatting. Any code styling issues which cause a commit to fail will be reported by the tool in the terminal.

### PR DESCRIPTION
**Pull request by Shapokhan**
fixes #32

### CODE REVIEW

**Comment by Shapokhan** [comment]
Reference link: [https://www.inkoop.io/blog/setup-eslint-for-react-with-prettier-pre-commit-and-vscode/](url)

File: package.json [hunk 1]
```diff
@@ -10,13 +10,17 @@
...
+    "yarn": "^1.22.19"
   },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "lint": "eslint ./",
+    "lint-fix": "eslint ./ --fix",
```
Comment (omeryousaf) [inline package.json:22]: @Shapokhan what's the difference between `lint-fix` and `format` commands ?

File: package.json [hunk 2]
```diff
@@ -10,13 +10,17 @@
...
+    "format": "prettier --write \"**/*.{js,jsx,json,md}\""
```
Comment (omeryousaf) [inline package.json:23]: @Shapokhan what's the difference between `lint-fix` and `format` commands ?

Comment (Shapokhan) [reply package.json:23]: @omeryousaf bhai, lint-fix is eslint command used to identify linting errors like unused variables, missing semicolons, incorrect spacing, etc, and automatically fix them. 
Format is a prettier command that only focuses on styling and formate like indentation, line wrapping, code alignment etc.

**Review by omeryousaf — APPROVED** [review]

**Comment by omeryousaf** [comment]
@Shapokhan it seems `husky` requires additional installation script after being added to the project's `node_modules`. See their docs: https://typicode.github.io/husky/getting-started.html#manual.

**Comment by omeryousaf** [comment]
@Shapokhan according to husky's guide, the way to add a command that we want husky to run, is different from what the tutorial you followed suggests. Here is what the docs suggest: https://typicode.github.io/husky/getting-started.html#create-a-hook.

File: package.json [hunk 3]
```diff
@@ -32,5 +36,26 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
+    "eslint-config-airbnb": "^19.0.4",
+    "eslint-config-prettier": "^8.8.0",
+    "eslint-plugin-import": "^2.27.5",
```
Comment (omeryousaf) [inline package.json:43]: @Shapokhan if you switch to another branch and try to run the project without reinstalling package.json, do you also see an error like the following in ur terminal ?
image

Comment (omeryousaf) [reply package.json:43]: It seems to suggest that `eslint` is already a dependency of react-scripts, so our project already has eslint installed when we install react-scripts.

Comment (omeryousaf) [reply package.json:43]: Some guidance from react docs on how to configure eslint: https://create-react-app.dev/docs/setting-up-your-editor/#extending-or-replacing-the-default-eslint-config.

Comment (Shapokhan) [reply package.json:44]: @omeryousaf i was facing some errors. when i google it, someone suggests me to upgrade the eslint version manually. My error was gone and i can run the project smoothly without any issue. But it seems to cause a problem on your side.

Comment (Shapokhan) [reply package.json:44]: > Some guidance from react docs on how to configure linting: https://create-react-app.dev/docs/setting-up-your-editor/#formatting-code-automatically

let me try this configuration of eslint.

Comment (omeryousaf) [reply package.json]: @Shapokhan which eslint version did you manually install ?

Comment (Shapokhan) [reply package.json]: I was using the version "eslint": "^8.42.0". It was working fine at that time but when i restart visual studio code. Now it says.
Screenshot 2023-06-09 at 12 32 57 PM

File: package.json [hunk 4]
```diff
@@ -7,16 +7,25 @@
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.3.2",
     "@testing-library/user-event": "^7.1.2",
+    "husky": "^8.0.3",
```
Comment (omeryousaf) [inline package.json:10]: @Shapokhan should it be a dependency or devDependency ? Same for the 2 other libraries added below.

File: package.json [hunk 5]
```diff
@@ -32,5 +41,14 @@
...
     ]
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "devDependencies": {
+    "husky": "^8.0.0",
+    "pretty-quick": "^3.1.3"
```
Comment (omeryousaf) [inline package.json:52]: @Shapokhan do we need both `lint-staged` and `pretty-quick` or only one of them ?

Comment (Shapokhan) [reply package.json:40]: @omeryousaf i have removed lint-staged, only pretty-quick is enough to format the document before commit.

File: package.json (hunk 4 above)
Comment (Shapokhan) [reply package.json]: @omeryousaf yes, it should be in devDependency. You can review it again. I have removed it from devDependency.

**Review by omeryousaf — APPROVED** [review]

**Pull request merged**